nixos/modules/services/x11/display-managers/default.nix

```diff
@@ -194,9 +194,12 @@ let
         ${xorg.lndir}/bin/lndir ${pkg}/share/xsessions $out/share/xsessions
       '') cfg.displayManager.extraSessionFilePackages}
 
-      mkdir -p "$out/share/wayland-sessions"
+      
       ${concatMapStrings (pkg: ''
-        ${xorg.lndir}/bin/lndir ${pkg}/share/wayland-sessions $out/share/wayland-sessions
+        if test -d ${pkg}/share/wayland-sessions; then
+          mkdir -p "$out/share/wayland-sessions"
```

Comment:
Why are we creating the directory? Should not `lndir` already do that?

Reply:
Nope, looks like `lndir fromdir todir` will fail if `todir` doesn't exist already.